rest-service/manager_rest/rest/resources_v1/events.py

```diff
@@ -337,6 +337,7 @@ def select_column(column_name):
                 select_column('level'),
                 literal_column("'cloudify_{}'".format(model.__name__.lower()))
                 .label('type'),
+                func.count('*').over().label('full_count'),
```

Comment:
Why is adding this to each result better than running a single query?

Reply:
Because we don't run a single query. What we currently do is:
1. Run a counting query (which differs enough from the main query to get different total results when some filters are used), using its result as the total.
2. Run the main query, which then obtains the amount of results specified from the defined offset.

Unfortunately, as I say, they give different results even without the limit and offset, because they are massively different queries.
Changing to do it with one query guarantees that we return specifically the total possible results at the time of this query.

In terms of efficiency of data transfer I don't think it'll have that great an effect- it's one extra integer (or possibly long), so at worst that should be 4 or 8 bytes. Every record returned already has at least seven bytes consumed by the 'cloudify_log' (or more for the event) type string, plus the message, the timestamps, the workflow ID, the blueprint ID, the operation, etc, etc, most of which are strings which means that we should be looking at a minimal increase in data transferred in exchange for getting consistent results and only making a single query each time (rather than two for every list request against the REST API).

Reply:
If we have the actual query that's getting the results, can't we do `query.count()` instead of `query.all()` to get the total (maybe excluding offset/limit params)?

Reply:
The problem is that we need to use the offset/limit if we want to actually efficiently get a restricted set of results, otherwise we're going to load every single result into memory in the rest service every time, then slice them (taking a bit more memory) then convert them to json (taking even more memory), which potentially gets us back to the exact same problem we had before.

Reply:
(the same problem that required us to introduce pagination, that is)